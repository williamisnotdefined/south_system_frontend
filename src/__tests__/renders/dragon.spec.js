import React from 'react'
import { render, wait } from '@testing-library/react'
import { ThemeProvider } from 'styled-components'

import { theme } from '@styles'
import ContextContainer from '@containers'
import DragonForm from '@pages/Dragon'

const formEdit = () => (
    <ThemeProvider theme={theme}>
        <ContextContainer>
            <DragonForm match={{ params: { id: '0' } }} />
        </ContextContainer>
    </ThemeProvider>
)

const formNew = () => (
    <ThemeProvider theme={theme}>
        <ContextContainer>
            <DragonForm match={{ params: { id: null } }} />
        </ContextContainer>
    </ThemeProvider>
)

describe('Testing Dragon Form Page Render', () => {
    it('should render correctly edit form after load dragon', async () => {
        const form = render(formEdit())
        await wait(() => expect(form).toMatchSnapshot())
    })

    it('should render correctly create form after load dragon', async () => {
        const form = render(formNew())
        await wait(() => expect(form).toMatchSnapshot())
    })
})
