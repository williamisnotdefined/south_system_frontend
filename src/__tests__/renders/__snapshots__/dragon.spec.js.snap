// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Dragon Form Page Render should render correctly create form after load dragon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxheI bzmUgQ"
      >
        <h1
          class="sc-AxjAm gwODik"
        >
          Creating Dragon
        </h1>
        <form
          class="sc-AxhUy iwSbOU"
        >
          <input
            autocomplete="off"
            class="sc-AxhCb hMENDW"
            name="name"
            placeholder="Type dragon name.."
            type="text"
            value=""
          />
          <input
            autocomplete="off"
            class="sc-AxhCb hMENDW"
            name="kind"
            placeholder="Type dragon kind.."
            type="text"
            value=""
          />
          <button
            class="sc-AxgMl fgEWzF"
            type="submit"
          >
            Entrar
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxheI bzmUgQ"
    >
      <h1
        class="sc-AxjAm gwODik"
      >
        Creating Dragon
      </h1>
      <form
        class="sc-AxhUy iwSbOU"
      >
        <input
          autocomplete="off"
          class="sc-AxhCb hMENDW"
          name="name"
          placeholder="Type dragon name.."
          type="text"
          value=""
        />
        <input
          autocomplete="off"
          class="sc-AxhCb hMENDW"
          name="kind"
          placeholder="Type dragon kind.."
          type="text"
          value=""
        />
        <button
          class="sc-AxgMl fgEWzF"
          type="submit"
        >
          Entrar
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Testing Dragon Form Page Render should render correctly edit form after load dragon 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-AxheI bzmUgQ"
      >
        <h1
          class="sc-AxjAm gwODik"
        >
          Editing Dragon
        </h1>
        <form
          class="sc-AxhUy iwSbOU"
        >
          <input
            autocomplete="off"
            class="sc-AxhCb hMENDW"
            name="name"
            placeholder="Type dragon name.."
            type="text"
            value="Green Dragon"
          />
          <input
            autocomplete="off"
            class="sc-AxhCb hMENDW"
            name="kind"
            placeholder="Type dragon kind.."
            type="text"
            value="Green"
          />
          <input
            autocomplete="off"
            class="sc-AxhCb hMENDW"
            disabled=""
            name="createAt"
            type="text"
            value="5/24/2020 6:45:33 AM"
          />
          <button
            class="sc-AxgMl fgEWzF"
            type="submit"
          >
            Entrar
          </button>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-AxheI bzmUgQ"
    >
      <h1
        class="sc-AxjAm gwODik"
      >
        Editing Dragon
      </h1>
      <form
        class="sc-AxhUy iwSbOU"
      >
        <input
          autocomplete="off"
          class="sc-AxhCb hMENDW"
          name="name"
          placeholder="Type dragon name.."
          type="text"
          value="Green Dragon"
        />
        <input
          autocomplete="off"
          class="sc-AxhCb hMENDW"
          name="kind"
          placeholder="Type dragon kind.."
          type="text"
          value="Green"
        />
        <input
          autocomplete="off"
          class="sc-AxhCb hMENDW"
          disabled=""
          name="createAt"
          type="text"
          value="5/24/2020 6:45:33 AM"
        />
        <button
          class="sc-AxgMl fgEWzF"
          type="submit"
        >
          Entrar
        </button>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
